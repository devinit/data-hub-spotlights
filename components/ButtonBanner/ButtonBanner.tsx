import React, { FunctionComponent } from 'react';
import { AnchorButton } from '../AnchorButton';
import { SpotlightBanner, SpotlightBannerAside } from '../SpotlightBanner';

interface ButtonBannerProps {
  onClick?: () => void;
  className?: string;
}

const ButtonBanner: FunctionComponent<ButtonBannerProps> = ({ onClick, children, className }) => (
  <SpotlightBanner>
    <SpotlightBannerAside>
      <AnchorButton className={className} onClick={onClick}>
        {children}
      </AnchorButton>
    </SpotlightBannerAside>
  </SpotlightBanner>
);

export { ButtonBanner };
