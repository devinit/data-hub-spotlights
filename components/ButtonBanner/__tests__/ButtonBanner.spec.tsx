/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { ButtonBanner } from '../ButtonBanner';

describe('ButtonBanner', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<ButtonBanner />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders the provided className', () => {
    const renderer = TestRenderer.create(<ButtonBanner className="button-banner" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <ButtonBanner>
        <span>Caption</span>
      </ButtonBanner>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
