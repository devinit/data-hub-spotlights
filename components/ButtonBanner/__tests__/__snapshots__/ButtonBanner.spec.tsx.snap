// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBanner renders correctly 1`] = `
<div
  className="spotlight-banner"
>
  <div
    className="spotlight-banner__aside"
  >
    <a
      className="button"
    >
      <style
        jsx={true}
      >
        
      cursor: pointer;
    
      </style>
    </a>
  </div>
</div>
`;

exports[`ButtonBanner renders the provided className 1`] = `
<div
  className="spotlight-banner"
>
  <div
    className="spotlight-banner__aside"
  >
    <a
      className="button-banner"
    >
      <style
        jsx={true}
      >
        
      cursor: pointer;
    
      </style>
    </a>
  </div>
</div>
`;

exports[`ButtonBanner renders with children 1`] = `
<div
  className="spotlight-banner"
>
  <div
    className="spotlight-banner__aside"
  >
    <a
      className="button"
    >
      <style
        jsx={true}
      >
        
      cursor: pointer;
    
      </style>
      <span>
        Caption
      </span>
    </a>
  </div>
</div>
`;
